sdk/azcore/policy_body_download.go

```diff
@@ -22,17 +22,18 @@ func newBodyDownloadPolicy() Policy {
 			return resp, err
 		}
 		var opValues bodyDownloadPolicyOpValues
-		if req.OperationValue(&opValues); !opValues.skip && resp.Body != nil {
-			// Either bodyDownloadPolicyOpValues was not specified (so skip is false)
-			// or it was specified and skip is false: don't skip downloading the body
-			var b []byte
-			b, err = ioutil.ReadAll(resp.Body)
-			resp.Body.Close()
-			if err != nil {
-				return resp, newBodyDownloadError(err, req)
-			}
-			resp.Body = &nopClosingBytesReader{s: b}
+		// don't skip downloading error response bodies
+		if req.OperationValue(&opValues); opValues.skip && resp.StatusCode < 400 {
+			return resp, err
+		}
+		// Either bodyDownloadPolicyOpValues was not specified (so skip is false)
+		// or it was specified and skip is false: don't skip downloading the body
+		b, err := ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+		if err != nil {
+			return resp, newBodyDownloadError(err, req)
```

Comment:
Why do we return the response here and not nil?

Reply:
In case other stages in the pipeline need to inspect the response (e.g. logging).  The only time we should be returning a nil response in the pipeline is if the response is in fact nil.

Reply:
Got it, thanks! :)